lib/ansible/playbook/task.py

```diff
@@ -106,6 +107,90 @@ def __init__(self, block=None, role=None, task_include=None):
 
         super(Task, self).__init__()
 
+    def _validate_module_defaults(self, attribute, name, value):
+        # load, resolve, and cache any action groups the task may use
+        if value is None:
+            return
+
+        for defaults_dict in value:
+            for defaults_entry, defaults in defaults_dict.items():
+                if not defaults_entry.startswith('group/'):
+                    continue
+
+                group_name = defaults_entry.split('group/')[-1]
+                collection_name = '.'.join(group_name.split('.')[0:2])
+                self._resolve_group(group_name, collection_name)
+
+        # _load_module_defaults in the base class ensures value is a list of dictionaries,
+        # the keys of which are fully qualified action and group names
+        setattr(self, name, value)
+
+    def _resolve_group(self, group, collection_name):
+        # The group should be part of the current collection
+        if not group.startswith(collection_name + '.'):
+            fq_group_name = collection_name + '.' + group
+        else:
+            fq_group_name = group
+
+        # Check if the group has already been resolved and cached
+        if fq_group_name in self._parent._play._group_actions:
+            return fq_group_name, self._parent._play._group_actions[fq_group_name]
+
+        try:
+            if collection_name == 'ansible.legacy':
+                action_groups = _get_collection_metadata('ansible.builtin').get('action_groups', {})
+            else:
+                action_groups = _get_collection_metadata(collection_name).get('action_groups', {})
+        except ValueError:
+            # Don't fail if the collection isn't installed
+            return fq_group_name, []
+
+        # The collection may or may not use the fully qualified name
+        # Don't fail if the group doesn't exist in the collection
+        short_name = fq_group_name.split(collection_name + '.')[-1]
+        action_group = action_groups.get(
+            fq_group_name,
+            action_groups.get(short_name, [])
+        )
+
+        resolved_actions = []
+        for action in action_group:
+            # Check if this is a special 'metadata' entry
+            if isinstance(action, dict) and len(action) == 1 and 'metadata' in action:
+                for extend_group in action['metadata'].get('extend_group', []):
```

Comment:
maybe coerce a non-list value to a list? eg, allow:
`extend_group: some.other.group`